docs/software-development-kits/java/middleware.mdx

```diff
@@ -0,0 +1,133 @@
+
+And them execute:
+```bash
+mvn clean package
+```
+
+In order for Spring to discovery the filter, we need to add the `@ComponentScan` annotation to the main class of the application.
+```java
+@ComponentScan("com.aserto")
+```
+
+## Configuration
+
+The authorization filter accepts the fallowing configuration parameters:
+
+| Parameter name | Default Value | Description |
+| -------------- | ------------- | ----------- |
+| aserto.authorization.enabled | true | Enable/disable the authorization filter |
+| aserto.authorization.host | localhost | The hostname of the authorizer service |
+| aserto.authorization.port | 8282 | The port of the authorizer service |
+| aserto.authorizer.insecure | false | Use insecure connection to the authorizer service |
+| aserto.authorizer.cert.path | N/A | The path to the authorizer service certificate, usefull when using local topaz with self signed certificates |
+| aserto.authorizer.apiKey | N/A | The API key to use when connecting to the authorizer service |
+| aserto.tenantId | N/A | The tenant ID to use when connecting to the authorizer service |
+| aserto.authorizer.policyName | N/A | The policy name used by the authorizer |
+| aserto.authorizer.policyLabel | N/A | The policy label used by the authorizer |
+| aserto.authorizer.decision | N/A | The decision used by the authorizer |
+| logging.level.com.aserto | N/A | The log level for the Aserto Middleware |
+
+
+## Identity
+To determine the identity of the user, the middleware can be configured to use a JWT token or a claim using the IdentityMapper.
+All you have to do is create a bean that return a InstanceMapper interface and the middleware will use it to determine the identity of the user.
+
+E.g.
+- use a JWT token from the `Authorization` header:
+```java
+    @Bean
+    public IdentityMapper identityDiscoverer() {
+        Extractor headerExtractor = new HeaderExtractor("Authorization");
+        return new JwtIdentityMapper(headerExtractor);
+    }
+```
+
+- use the "sub" claim from the JWT token in the `Authorization` header:
+```java
+    @Bean
+    public IdentityMapper identityDiscoverer() {
+        Extractor headerExtractor = new AuthzHeaderExtractor("Authorization", "sub");
+        return new SubjectIdentityMapper(headerExtractor);
+    }
+```
+
+You can also implement your own IdentityMapper by implementing the `IdentityMapper` interface.
+
+## Policy path
+By default, when computing the policy path, the middleware:
+* converts all slashes to dots
+* converts any path parameters to `__<param_name>`
+* converts uppercase characters in the URL path to lowercase
+
+As with the identity contest, you can implement your own PolicyPathMapper by implementing the `PolicyPathMapper`
+interface and creating a bean that returns it.
+
+E.g.
+```java
+    @Bean
+    public PolicyMapper policyMapperDiscoverer() {
+        return new MyCustomPolicyPathMapper(handlerMapping);
```

Comment:
What is `handlerMapping`? It's not defined in this snippet/

Reply:
This is a leftover from the `HttpPathPolicyMapper`. Removed the argument from the constructor as the number and type of arguments will be decided by the one that writes the `MyCustomPolicyPathMapper ` constructor.